soh/soh/Enhancements/Fixes/BrokenGiantsKnife.cpp

```diff
@@ -39,9 +35,9 @@ void OnReceiveBrokenGiantsKnife(GetItemEntry itemEntry) {
     }
 }
 
-void RegisterBrokenGiantsKnifeFix() {
+static void RegisterBrokenGiantsKnifeFix() {
     // If enhancement is off, flag should be handled exclusively by vanilla behaviour
-    COND_HOOK(OnItemReceive, CVAR_BGS_FIX_VALUE || IS_RANDO, OnReceiveBrokenGiantsKnife);
+    COND_ID_HOOK(OnItemReceive, ITEM_SWORD_BGS, CVAR_BGS_FIX_VALUE || IS_RANDO, OnReceiveBrokenGiantsKnife);
```

Comment:
seems a bit overkill to extend using ID hooks beyond actor ID for one hook implementation

Reply:
Yeah, we probably don’t want this, as an item from a mod (namely randomizer) will have a conflict here with ITEM_SWORD_BGS, while not actually being bgs